pkg/provider/azure_loadbalancer_backendpool.go

```diff
@@ -924,3 +924,88 @@ func removeNodeIPAddressesFromBackendPool(
 
 	return changed
 }
+
+type backendPoolTypePodIP struct {
+	*Cloud
+}
+
+func newBackendPoolTypePodIP(c *Cloud) BackendPool {
+	return &backendPoolTypePodIP{c}
+}
+
+func (bpi *backendPoolTypePodIP) CleanupVMSetFromBackendPoolByCondition(_ context.Context, _ *armnetwork.LoadBalancer, _ *v1.Service, _ []*v1.Node, _ string, _ func(string) bool) (*armnetwork.LoadBalancer, error) {
```

Comment:
where is the lb part? we'd need first provision lb and then create/update backend pool